lib/jobsub_api.py

```diff
@@ -70,7 +91,530 @@ def jobsub_call(argv: List[str], return_output: bool = False) -> Optional[str]:
+            else:
+                args.append(str(kwargs[k]))
+        elif k in jobsub_flags:
+            args.append(jobsub_flags[k])
+        else:
+            raise TypeError(f"unknown argument {k}")
+
+    args.append(f"file://{executable}")
+    args.extend(exe_arguments)
+
+    rs = jobsub_call(args, True)
+    m = jobsub_submit_re.search(rs)
+    if m:
+        job = SubmittedJob(
+            kwargs["group"],
+            m.group("jobid"),
+            kwargs.get("pool", ""),
+            kwargs.get("auth_methods", ""),
+            kwargs.get("role", ""),
+            rs,
+        )
+        # we think we know some jobsub_q attrs due to having just launched
+        job.set_q_attrs(
+            os.environ["USER"],
+            str(int(time.time())),
+            "0.0",
+            "I",
+            executable,
+            "0.0",
+            "0.0",
+        )
+        return job
+    raise JobsubAPIError(f"submission failed: {rs}")
+
+
+def jsq_date_to_datetime(s: str) -> datetime:
+    """convert either jobsub_q start date or --long datestamp to datetime"""
+    if s.find("/") > 0:
+        # near the year boundary the start date may be from last year
+        # indicated by a month larger than the current month (i.e
+        # start date month is in December but it is January
+        now = datetime.now()
+        sdmonth = int(s[:2])
+        year = now.year
+        if sdmonth > now.month:
+            year = year - 1
+        # convert to full iso date string dashes for month-day, T before time
+        isos = s.replace(" ", "T").replace("/", "-")
+        # and year in front
+        isos = f"{year}-{isos}"
+        return datetime.fromisoformat(isos)
+    return datetime.fromtimestamp(int(s))
+
+
+def jsq_runtime_to_timedelta(s: str) -> timedelta:
+    """convert jobsub_q runtime or --long float value to timedelta"""
+    if s.find(":") > 0:
+        days, hours, minutes, seconds = re.split("[+:]", s)
+        return timedelta(
+            days=int(days), hours=int(hours), minutes=int(minutes), seconds=int(seconds)
+        )
+    return timedelta(seconds=int(float(s)))
+
+
+JSMAP: Dict[str, JobStatus] = {}
+
+
+def init_JSMAP() -> None:
+    """make a map that converts status strings to JobStatus values
+    from either jobsub_q status leters IRXH or --long 01245 values
+    by going through the htcondor.JobStatus enums class
+    """
+    if len(JSMAP.keys()):
+        return
+    for k in JobStatus.__members__:
+        if k == "REMOVED":
+            jl = "X"
+        else:
+            jl = k[0]
+        if jl not in JSMAP:
+            JSMAP[jl] = JobStatus.__members__[k]
+            JSMAP[str(int(JobStatus.__members__[k]))] = JobStatus.__members__[k]
+
+
+qargs = [
+    "group",
+    "auth_methods",
+    "global_pool",
+    "role",
+    "subgroup",
+    "constraint",
+    "name",
+    "user",
+]
+
+
+def q(
+    *jobids: str, devserver: bool = False, verbose: int = 0, **kwargs: str
```

Comment:
Same comment here as with `submit` - that we should have `group` be an explicit argument here, to communicate that we expect it to be provided somehow.